### DISCUSSION TITLE
Use of the axial gauge for a QCD calculation

### DISCUSSION THREAD

**Discussion by vbertone** [FeynCalc/feyncalc#186] [Q&A] [answered]
Hello,

I'm running FeynCalc 10.0.0 with FeynArts 3.11 and FeynHelpers 1.4.0 on a Mac using Mathematica 12.3.1.0.

I need to run a one-loop (to be possibly extended to two loops) computation in QCD and I would like to use the axial gauge.
In fact, I would like to use the light-cone gauge but I guess this can be settled afterwards.
I mean to use the usual chain: FeynArts for diagrams and amplitudes generation and FeynCalc for the evaluation.
As far as I can see from here:

https://feyncalc.github.io/FeynCalcBookDev/GluonPropagator.html

FeynCalc supports the axial gauge.
So the first of my questions is: how can I generate amplitudes in the axial gauge?

The second question is more convoluted.
Is it possible to insert an external constraint in the evaluation of the loop integrals?
More precisely, what I need to do is something like this:
$$\int d^{4-2\epsilon}k_1 d^{4-2\epsilon}k_2\dots \mathcal{A}(k_1,k_2,\dots) {\color{red}\delta\left(x-\sum_{i}k_i^+\right)},\qquad k_i^+=k_{i,0}+k_{i,3}$$
where $\mathcal{A}$ is the "unintegrated" amplitude and in red I have highlighted my external constraint that is just a 1D $\delta$-function that acts on the +-component of the loop momenta, with $x$ some constant.

I hope I stated my questions clearly enough.
Looking forward to some feedback. Thank you.

Valerio

**Comment by vsht** [comment, marked as answer]
Hi,

as to your first question, GluonPropagator[] is actually very
misleading here. Yes, there are this (and few other) functions
that are meant for people entering amplitudes by hand.

Despite of the fact that this stuff was in FeynCalc since
the 90s, I don't think that anyone used it for anything beyond tree-level.
Simply because only since version 9.3 FeynCalc sort of can deal with
eikonal propagators in loops.

However, in your case the problem would be FeynArts, that doesn't
support axial gauage. I guess you could hack the SM model and replace
the propagator by hand or do it on the amplitude level, though.

Of course, having eikonal propagators also means that you will
have to calculate all masters by yourself, since there are no
libraries of analytic results for that.

Regarding your second question, such a measurement function
can be in principle added, but I don't think that FeynCalc
alone would be of much help here. Simply because FeynCalc doesn't
evaluate integrals analytically/numerically.

Personally, here I would use reverse unitarity to rewrite the
Delta as an eikonal propagator and then put the integrals
into FIRE/KIRA to get the IBPs done. Then probably use differential
equations with Canonica/Libra/whatever to calculate the masters.
I suppose that one could use FeynCalc for this or that,
but the bulk of the calculation would require completely different tools.

Cheers,
Vladyslav

**Comment by vbertone** [comment]
Hi Vladyslav,

thanks for your prompt and exhaustive answer.
This clears up my path.
The bottomline is that what I have in mind cannot be straightforwardly automated.
I may get back to you with other questions.

Cheers,
Valerio

**Reply by vsht** [reply]
Hi Valerio,

well, I actually didn't mean to phrase it in such a pessimistic way. It certainly can be 
automated, but you will need to invest "somewhat" more effort as compared to cases
where all integrals can be mapped to PaVe-functions.

It also depends a lot on the number of scales involved and whether you are aiming
for an analytic or a numerical solution.

The axial gauge propagator is quite easy to implement on the FeynArts level. One can
just use a modified Lorentz.gen  (say LorentzAxial.gen) and then do something like

```
diags = InsertFields[CreateTopologies[0, 2 -> 2], {V[5], V[5]}->
		{F[3, {1}], -F[3, {1}]}, InsertionLevel -> {Classes},
		Model -> "SMQCD",GenericModel->FileNameJoin[{$FeynArtsDirectory,"Models","LorentzAxial"}]];

Paint[diags, ColumnsXRows -> {2, 1}, Numbering -> Simple,
	SheetHeader->None,ImageSize->{512,256}];


amp[0] = FCFAConvert[CreateFeynAmp[diags], IncomingMomenta->{p1,p2},
	OutgoingMomenta->{k1,k2},UndoChiralSplittings->True,ChangeDimension->4,
	TransversePolarizationVectors->{p1,p2}, List->True, SMP->True,
	Contract->True,DropSumOver->True][[1]]/.{
	gluonProp[V[Index[Generic, 5]]]->1,
	gluonProp[x/;x=!=V[Index[Generic, 5]]]->0,
	genVecBosonProp[V[Index[Generic, 5]]]->0,
	genVecBosonProp[x/;x=!=V[Index[Generic, 5]]]->1
```	

Roughly speaking, the delta function will probably need an auxiliary vector ($n_2 = (1,0,0,-1)$?) so that one can rewrite it as  $\delta (x - k \cdot n_2)$. Which, by using reverse unitarity ([1](https://arxiv.org/abs/hep-ph/0207004), [2](https://arxiv.org/abs/hep-ph/0306192)), can be presumably treated as another eikonal propagator $\sim \frac{1}{x - k \cdot n_2}$ (ignoring the difference with different $i \varepsilon$ here).

Once one arrives at the final set of master integrals, one could undo the reverse unitarity and try to calculate the integrals
explicitly.



[LorentzAxial.zip](https://github.com/FeynCalc/feyncalc/files/9200562/LorentzAxial.zip)

**Comment by vbertone** [comment]
Hi Vladyslav,

thank you so much for the further insight!
Actually, your first message did not sound that negative, just realistic and it already gave me quite some elements to work on.

In fact, I was moving precisely along that direction, i.e. trying to define a new generic model in FeynArts but I was having some difficulties that you just solved. :-)

I found the idea of reverse unitarity illuminating.
Googleing after our first exchange, I actually came across the first of the papers you pointed me to plus a couple of presentations by Anastasiou where I found some interesting details. I'm trying to pursue that way.

As for the integrals with the eikonal propagator, I found [this](https://inspirehep.net/literature/133121) old paper that is good for the one loop integrals. Perhaps moving from here I can derive something useful at two loops.

Just to put my question into context and to motivate why I decide to go for the axial gauge, let me explain a little more in detail what I would like to do.
My goal is the computation of the two-loop anomalous dimensions of the so-called generalised parton distributions (GPDs), that are a generalisation of parton distribution functions that I think you are familiar with.
I recently published the one-loop [calculation](https://inspirehep.net/literature/2091328) where I did the calculation essentially by hand.
Of course, at two loops this is non affordable.
The basic problem here is that GPDs come with a Wilson line and if one works in Lorentz gauge one also needs to include explicit Feynman rules for the eikonal vertices and propagators.
In light-cone gauge instead the Wilson line unitarises and there is no need to care about Wilson lines and ghosts, which to me looks a substantial simplification.

Thanks again for your very useful input!

Cheers,
Valerio

**Reply by vsht** [reply]
Hi Valerio,

> As for the integrals with the eikonal propagator, I found this old paper that is good for the one loop integrals. Perhaps moving from here I can derive something useful at two loops.

if you expect to end up with integrals similar to what is given in the appendix of that paper, then I don't think that there
will be many difficulties calculating them. At one loop those are of course highly trivial and can be even done within
Mathematica like in the attached notebook 

[EikInts.zip](https://github.com/FeynCalc/feyncalc/files/9233418/EikInts.zip)

At two loops one might need HyperInt, but it may also be that the integrals are already known or again can be 
done purely in Mathematica.

BTW, it seems to me that there is a typo in that paper and the exponent should be $- i \pi/2$ instead of $- i N \pi/2$.  You can compare 
e.g. with  [Kleinert's book](http://users.physik.fu-berlin.de/~kleinert/kleiner_reb8/psfiles/phi4.pdf), Eq. 8.12

Cheers,
Vladyslav

**Reply by vbertone** [reply]
Hello Vladyslav,

thanks for the response of apologies for the slow reaction, I was on vacation.
As for the typo, I think that the factor $e^{-iN\pi/2}$ in Pritchard-Stirling's paper is correct.
The difference with Kleinert's must be that $\mathbf{q}\rightarrow -\mathbf{l}$, even though I didn't do the algebra to ascertain it.

I think that the two-loop eikonal integrals are known (after all Curci, Furmansky, and Petronzio computed the two-loop splitting functions in light-cone gauge more than 40 years ago). I did short survey but could not find them but I'm confident they exist.

Cheers,
Valerio

**Reply by vsht** [reply]
Hi,

> The difference with Kleinert's must be that q→−l, even though I didn't do the algebra to ascertain it.

I don't think that this is the reason for the discrepancy. You can always do a trivial sign switch of the loop momentum 
$k \to -k$ without changing the result. Effectively, this will also change the sign of the external momentum in the propagator.
Then one sees that both integrands are identical. Unless the paper uses an implicit overall normalization, I would expect that
Kleinert's result multiplied by $(2 \pi)^D$ should give the l.h.s of A.1.

Cheers,
Vladyslav

**Reply by vbertone** [reply]
Hi Vladyslav,

you're right!
In addition the result must only depend on $q^2$ which of course leaves the result unchanged upon sign change.
I also checked the dim. reg. [paper](https://reader.elsevier.com/reader/sd/pii/0550321372902799?token=B5481CF81877C9BE15710EE41A1A78F8B3AF9D19524F8EA7D6F27766B9CD3DDCD3BCDA711D3E1855624864BF53718923&originRegion=eu-west-1&originCreation=20220816143529) by Veltman and t'Hooft that they themselves cite (see e.g. Eq. (A.5)) and indeed that phase isn't there.

That's funny because they consisntently have that $e^{-iN \pi /2}$ all across the board.

Cheers,
Valerio

**Reply by vsht** [reply]
Hi Valerio,

it is not uncommon to encounter papers with weird normalization conventions that are not explicitly spelled out in the text. What matters is that if you want to use their results, you have to be 100% sure that you understand the normalization. For integrals that you are not able to calculate by yourself, this check mostly amounts to a numerical evaluation that should be compared with the given literature result.

Personally, I mostly use pySecDec via my [interface](https://feyncalc.github.io/FeynHelpersBookDev/PSDCreatePythonScripts.html) to FeynCalc, although sometimes one encounters integrals that pySecDec cannot handle. Then I switch to FIESTA. For asymptotic expansions FIESTA is currently much better than pySecDec, but that might change in the future.

Cheers,
Vladyslav

**Comment by vbertone** [comment]
Hi Vladyslav,

I hope you are well.
I'm getting back to you because a collaborator of mine and I are trying to progress on the two-loop calculation that I briefly described above.
We have managed to reduce the complexity of the task to the computation of one-loop box diagrams.
However, these box diagrams are to be computed in a configuration that is perhaps not standard and, to the best of our understanding, does not allow us to use FeyCalc (+ Package-X) out of the box. However, we are confident that this can be done within FeynCalc using the right commands. Let me try to explain the problem.

One of the diagrams that we are dealing with is this one:

![](https://github.com/FeynCalc/feyncalc/files/10089986/qqbarPDFBox.pdf?raw=true)
where the arrows indicate the momentum flow. $p$ and $p'$ are on-shell and parallel, that is crucial as I will explain below, while $k$ is generally off the mass shell. Everything is massless.

What we need to do is to compute this diagrams (and a few others similar to this) with upper legs truncated (no wave functions), while the lower legs do carry a wave function and need to be summed over the spin indices such that, exploiting the fact that $p'=\alpha p$, $\alpha\in \mathbb{R}$, one has $\sum_s u(p,s)\overline{u}(p',s)=\sqrt{\alpha}\,\gamma\cdot p$. This procedure should result in a Dirac-indexed quantity that essentially depends on $kp$, $k^2$, and $\alpha$ (being $p^2=0$).
In addition, what we eventually need is not the complete integral but only its linear pole in the dim. reg. parameter $\epsilon$. In fact, this diagram probably has also a double pole that we do not actually need, but of course having it does not hurt. Do you think this calculation can be done in FeynCalc?

Another point regards the fact that, since we are working in light-cone gauge, the gluon propagators introduce the usual $1/(qn)$ terms that I believe cannot be handled at the moment within FeynCalc, but this is an issue that we can address at a later stage.

I hope I made my question clear enough. If not, please let me know if you need any further information.
Thank you in advance for your answer.

Cheers,
Valerio

**Comment by vsht** [comment]
Hi Valerio,

I don't think that this approach should cause issues within FeynCalc, since all ingredients appear to be there already.

You can have explicit Dirac indices using [DCHN](https://feyncalc.github.io/FeynCalcBookDev/DCHN.html), while linear propagators can be entered as [SFAD](https://feyncalc.github.io/FeynCalcBookDev/SFAD.html)s.

Perhaps you could try to see how far you can get and if there are issues or some nonobvious things, I could give my advice here and there.

Cheers,
Vladyslav

**Reply by vbertone** [reply]
Hello Vladyslav,

thanks to your suggestion I think I'm slowly progressing.

I now have a related question. Does TID know how to deal with eikonal propagators?
In other words, does the the tensor-integral decomposition w.r.t. a momentum, say $q$, return reliable result when denominators like $1/(n\cdot q)$ or even $1/(n\cdot(q+\dots))$ are present.
I'm asking because in this case TID[exp, q,, ToPaVe->True] returns results that depend on $q$ and this does not seem right.
Thank you in advance for your help.

Cheers,
Valerio

**Reply by vsht** [reply]
Hi,

well, for eikonal propagators one should better use `FCMultiLoopTID`, since these are integrals where there is no guarantee to remove numerators via tensor reduction. In the case of quadratic propagators this is always possible and `TID` was designed with this requirement in mind.

There is of course no harm in trying `TID`, but most likely the integral will be returned unevaluated:

```
TID[GSD[q] SFAD[{{0, q . n}}, {q, m^2}], q]
```
vs
```
FCMultiLoopTID[GSD[q] SFAD[{{0, q . n}}, {q, m^2}], {q}]
```

> I'm asking because in this case TID[exp, q,, ToPaVe->True] returns results that depend on and this does not seem right.

This is not surprising, since eikonal integrals cannot be expressed as PaVe functions. So `ToPaVe` / Package-X are useless here.
No matter how you arrive at your final set of master integrals, those have to be calculated by hand or extracted from the literature
and substituted into the amplitude via replacement rules.  I don't think that there is any publicly available library of analytic results for such integrals. For that matter, Package-X is probably the only existing library of exhaustive analytic results for loop integrals. People often take it for granted without realizing how unique this piece of software is. 

As a matter of fact, for your scenario one should probably not use `TID` / `FCMultiLoopTID` at all, but rather handle the calculation
using the new multi-loop functionality, i.e. topology identification, topology minimization, minimal tensor reduction with `FCLoopTensorReduce`, rewriting all integrals as `GLI`s, IBP reduction using FIRE, minimization of the masters etc.

You can have a look at the attached example and [arXiv:2112.14132](https://arxiv.org/pdf/2112.14132.pdf). This requires the latest 
dev versions of FeynCalc and FeynHelpers. The manuals  are here

https://github.com/FeynCalc/feyncalc-manual/releases/tag/dev-manual
https://github.com/FeynCalc/feynhelpers-manual/releases/tag/dev-manual

It can be edgy here and there, but the stuff is already pretty much usable, at least for things where I needed it in my calculations.

Cheers,
Vladyslav







[Gh-Gh-2L.zip](https://github.com/FeynCalc/feyncalc/files/10136275/Gh-Gh-2L.zip)
